reflex/app.py

```diff
@@ -701,6 +706,16 @@ def mark_complete(_=None):
                 stateful_components_code,
                 page_components,
             ) = compiler.compile_stateful_components(self.pages.values())
+
+            # Catch "static" apps (that do not define a rx.State subclass) which are trying to access rx.State.
+            if (
```

Comment:
Can we do a follow-up to abstract this out into a function `if uses_state(stateful_component_code)`